plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java

```diff
@@ -84,6 +81,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
       if (project != null) {
         Credential credential = loginIfNeeded(event);
         if (credential != null) {
+          new DeployPreferencesDialog(HandlerUtil.getActiveShell(event), project).open();
```

Comment:
I think you need to check the output of the dialog to see if a user pressed cancel. Currently, clicking the cancel button is like clicking the OK button, and it gives the following exception:

`org.eclipse.core.commands.ExecutionException: Project ID must be set for deploy.`